pkg/globalnet/controllers/ipam/ipam.go

```diff
@@ -607,6 +682,26 @@ func (i *Controller) podUpdater(obj runtime.Object, key string) error {
 	return nil
 }
 
+func (i *Controller) svcExUpdater(obj runtime.Object, key string) error {
+	var svcEx mcsv1a1.ServiceExport
+
+	unstructObj := obj.(*unstructured.Unstructured).UnstructuredContent()
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructObj, &svcEx)
+	if err != nil {
+		klog.Errorf("Failed to extract ServiceExport from unstructured Object %#v", unstructObj)
```

Comment:
This is likely an unrecoverable error so no reason to re-queue. However we don't really need to convert it. You only need to access `Namespace`  and `Name` which are easily available on `Unstructured`.

```suggestion
		klog.Errorf("Failed to extract ServiceExport from unstructured Object %#v", unstructObj)
		return nil
```

Reply:
Yeah, I have done a similar change. We don't even need Unstructured. We can get name+namespace from key itself.